GitUpKit/Core/GCRepository.m

```diff
@@ -306,7 +306,14 @@ - (BOOL)exportBlobWithSHA1:(NSString*)sha1 toPath:(NSString*)path error:(NSError
 #if !TARGET_OS_IPHONE
 
 - (NSString*)pathForHookWithName:(NSString*)name {
-  NSString* path = [[self.repositoryPath stringByAppendingPathComponent:@"hooks"] stringByAppendingPathComponent:name];
+  NSString* hooksPath;
+  NSString* configHooksPath = [[self readConfigOptionForVariable:@"core.hooksPath" error:NULL] value];
+  if (configHooksPath.length > 0) {
+    hooksPath = [self.workingDirectoryPath stringByAppendingPathComponent:configHooksPath];
```

Comment:
@lucasderraugh I've found this is incorrect, because users can set an absolute path for `core.hooksPath`. I'll set this PR as draft until fix this.

Reply:
I've fixed this.